app/containers/MessageBox/RecordAudio.js

```diff
@@ -0,0 +1,157 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { TouchableOpacity } from 'react-native';
+import { activateKeepAwake, deactivateKeepAwake } from 'expo-keep-awake';
+import { Audio } from 'expo-av';
+import RNFetchBlob from 'rn-fetch-blob';
+
+import { CustomIcon } from '../../lib/Icons';
+import styles from './styles';
+import I18n from '../../i18n';
+import { themes } from '../../constants/colors';
+import { mode as playbackMode } from '../message/Audio';
+
+const RECORDING_MODE = {
+	allowsRecordingIOS: true,
+	playsInSilentModeIOS: true,
+	staysActiveInBackground: false,
+	shouldDuckAndroid: true,
+	playThroughEarpieceAndroid: false,
+	interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
+	interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX
+};
+
+const RECORDING_SETTINGS = {
+	android: {
+		extension: '.aac',
+		outputFormat: Audio.RECORDING_OPTION_ANDROID_OUTPUT_FORMAT_AAC_ADTS,
+		audioEncoder: Audio.RECORDING_OPTION_ANDROID_AUDIO_ENCODER_AAC,
+		sampleRate: 22050,
+		numberOfChannels: 1,
+		bitRate: 128000
+	},
+	ios: {
+		extension: '.aac',
+		audioQuality: Audio.RECORDING_OPTION_IOS_AUDIO_QUALITY_MIN,
+		sampleRate: 22050,
+		bitRate: 128000,
+		outputFormat: Audio.RECORDING_OPTION_IOS_OUTPUT_FORMAT_MPEG4AAC,
+		numberOfChannels: 2
+	}
+};
+
+const startRecordingAudio = async(instance, setRecordingStatus) => {
+	try {
+		const permissions = await Audio.getPermissionsAsync();
```

Comment:
You don't need this logic, you can just call `await Audio.requestPermissionsAsync();` directly, if it was previously granted, the status returns granted and the prompt of permission not will be showed.  :)

Reply:
Was previously doing that. When I click on the button and it asks for permission on android, and after the permission is granted, even when I'm not touching anymore, it starts recording. Didn't want the audio to record when this happens.

Reply:
Can you show me a GIF?

Reply:
![MP4_20200619_225013](https://user-images.githubusercontent.com/7759208/85163071-eae9c100-b27f-11ea-811d-ee1b422a2351.gif)

Reply:
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/7759208/85163301-45831d00-b280-11ea-9f0b-dde3291662b4.gif)

Reply:
Sure! I'll take a look when it is more done.
You can maintain with your logic. 😁